sdk/translation/Azure.AI.Translation.Document/src/DocumentTranslationOperation.cs

```diff
@@ -7,125 +7,164 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Azure.AI.Translation.Document.Models;
 using Azure.Core;
 using Azure.Core.Pipeline;
 
 namespace Azure.AI.Translation.Document
 {
     /// <summary> Tracks the status of a long-running operation for translating documents. </summary>
-    public class DocumentTranslationOperation : PageableOperation<DocumentStatusResult>
+    public class DocumentTranslationOperation : PageableOperation<DocumentStatusResult>, IOperation<AsyncPageable<DocumentStatusResult>>
     {
         // TODO: Respect retry after #19442
         private readonly TimeSpan DefaultPollingInterval = TimeSpan.FromSeconds(30);
 
-        /// <summary>Provides communication with the Translator Cognitive Service through its REST API.</summary>
         private readonly DocumentTranslationRestClient _serviceClient;
-
-        /// <summary>Provides tools for exception creation in case of failure.</summary>
         private readonly ClientDiagnostics _diagnostics;
+        private readonly OperationInternal<AsyncPageable<DocumentStatusResult>> _operationInternal;
+
+        private int? _retryAfterHeaderValue;
```

Comment:
Sure. I looked more and the idea was to expose this as a property as the service returns the `Retry-After` header so the user can do that instead of setting an arbitrary value time. The problem is that currently the service always returns `1`, so until they don't return a real value, no point on the property. So I deleted it

Reply:
Unless specified otherwise, `OperationInternal<T>` already uses the `Retry-After` header to calculate delays. It uses a [RetryAfterDelayStrategy](https://github.com/Azure/azure-sdk-for-net/blob/9b21c111fb3135e866c00651f0780b15906403b0/sdk/core/Azure.Core/src/Shared/RetryAfterDelayStrategy.cs#L39) internally.

Reply:
So many goodness from using OperationInternal!

If the user is manually polling though, they will need to access the property from the header of the request, so we wanted to help them by exposing the property in the `Operation` class